test/unit/svg/svg_parser_test.cpp

```diff
@@ -291,5 +554,235 @@ TEST_CASE("SVG parser") {
+    }
+
+    SECTION("SVG <gradient> with transformations")
+    {
+        //
+        std::string svg_name("./test/data/svg/gradient-transform.svg");
+        std::shared_ptr<mapnik::marker const> marker = mapnik::marker_cache::instance().find(svg_name, false);
+        REQUIRE(marker);
+        REQUIRE(marker->is<mapnik::marker_svg>());
+        mapnik::marker_svg const& svg = mapnik::util::get<mapnik::marker_svg>(*marker);
+        auto bbox = svg.bounding_box();
+        REQUIRE(bbox == mapnik::box2d<double>(1.0,1.0,799.0,599.0));
+        auto storage = svg.get_data();
+        REQUIRE(storage);
+
+        auto const& attrs = storage->attributes();
+        REQUIRE(attrs.size() == 3 );
+        REQUIRE(attrs[1].fill_gradient == attrs[2].fill_gradient);
+        REQUIRE(attrs[1].fill_gradient.get_gradient_type() == mapnik::RADIAL);
+        agg::trans_affine transform;
+        transform *= agg::trans_affine_translation(240,155);
+        REQUIRE(attrs[1].fill_gradient.get_transform() == transform);
+    }
+
+    SECTION("SVG <gradient> with xlink:href")
+    {
+        std::string svg_name("./test/data/svg/gradient-xhref.svg");
+        std::shared_ptr<mapnik::marker const> marker = mapnik::marker_cache::instance().find(svg_name, false);
+        REQUIRE(marker);
+        REQUIRE(marker->is<mapnik::marker_svg>());
+        mapnik::marker_svg const& svg = mapnik::util::get<mapnik::marker_svg>(*marker);
+        auto bbox = svg.bounding_box();
+        REQUIRE(bbox == mapnik::box2d<double>(20,20,460,230));
+        auto storage = svg.get_data();
+        REQUIRE(storage);
+
+        auto const& attrs = storage->attributes();
+        REQUIRE(attrs.size() == 2 );
+        REQUIRE(attrs[0].fill_gradient.get_gradient_type() == mapnik::LINEAR);
+        REQUIRE(attrs[1].fill_gradient.get_gradient_type() == mapnik::LINEAR);
+        REQUIRE(attrs[1].fill_gradient.has_stop());
+    }
+
+    SECTION("SVG <gradient> with radial percents")
+    {
+        std::string svg_name("./test/data/svg/gradient-radial-percents.svg");
+        std::shared_ptr<mapnik::marker const> marker = mapnik::marker_cache::instance().find(svg_name, false);
+        REQUIRE(marker);
+        REQUIRE(marker->is<mapnik::marker_svg>());
+        mapnik::marker_svg const& svg = mapnik::util::get<mapnik::marker_svg>(*marker);
+        auto bbox = svg.bounding_box();
+        REQUIRE(bbox == mapnik::box2d<double>(0,0,200,200));
+        auto storage = svg.get_data();
+        REQUIRE(storage);
+
+        double x1, x2, y1, y2, r;
+        auto const& attrs = storage->attributes();
+        REQUIRE(attrs.size() == 1 );
+        REQUIRE(attrs[0].fill_gradient.get_gradient_type() == mapnik::RADIAL);
+        REQUIRE(attrs[0].fill_gradient.has_stop());
+        attrs[0].fill_gradient.get_control_points(x1, y1, x2, y2, r);
+        REQUIRE(x1 == 0);
+        REQUIRE(y1 == 25);
+        REQUIRE(x2 == 10);
+        REQUIRE(y2 == 10);
+        REQUIRE(r == 75);
```

Comment:
Question, @artemp. The gradient fixture for this test has the following format
```
<radialGradient id="grad1b" gradientUnits="userSpaceOnUse" cx="10%" cy="10%" r="75%" fx="00.00%" fy="25%">
```
This translates to the above values when `get_control_points` is called. Is this the expected value? Should the values be `0, 0.25, 0.10, 0.10, 0.75` respectively due to the percentage conversion?